### PR TITLE
[Enhancement](delete) eliminate reading the old values of non-key columns for delete stmt

### PR DESCRIPTION
**Pull request by bobhan1**
## Proposed changes
Currently, when executing `insert into select` statement converted from a delete statement, doris will try to read all the values of the non-key columns from the previous rows with the same key in storage layer, which is highly costly and meaningless.
For example, for a table with the following schema
```
CREATE TABLE test (
    `k1` int NOT NULL,
    `c1` int,
    `c2` int,
    `c3` int,
    `c4` int)
UNIQUE KEY(`k1`) DISTRIBUTED BY HASH(`k1`) BUCKETS 1
PROPERTIES("enable_unique_key_merge_on_write" = "true")
```
and data
```
k1,c1,c2,c3,c4
1,1,1,1,1
2,2,2,2,2
3,3,3,3,3
4,4,4,4,4
5,5,5,5,5
```
after execute some delete statement(which is transformed to `insert into select` statement) that delete rows with k1=1,2,3, the data with delete_sign column will be like
```
k1,c1,c2,c3,c4,__DORIS_DELETE_SIGN__
1,1,1,1,1,0
2,2,2,2,2,0
3,3,3,3,3,0
4,4,4,4,4,0
5,5,5,5,5,0
1,1,1,1,1,1
2,2,2,2,2,1
3,3,3,3,3,1
```
It's  evitable that it reads lots of useless data. So this PR eliminates the above process and use default or null value to fill the non-key columns when inserting rows with marked delete sign, since the values in these columns are useless and will not be read.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.36 seconds
 stream load tsv:          506 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.6 seconds inserted 10000000 Rows, about 337K ops/s
 storage size: 17166804879 Bytes

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.4 seconds
 stream load tsv:          505 seconds loaded 74807831229 Bytes, about 141 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          29.0 seconds inserted 10000000 Rows, about 344K ops/s
 storage size: 17169927022 Bytes

**Comment by bobhan1** [comment]
run p0

**Comment by bobhan1** [comment]
run p0

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run arm

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.19 seconds
 stream load tsv:          509 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.4 seconds inserted 10000000 Rows, about 340K ops/s
 storage size: 17168762914 Bytes

**Comment by bobhan1** [comment]
run buildall

File: regression-test/suites/unique_with_mow_p0/partial_update/test_partial_update_delete.groovy [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    sql "insert into ${tableName1} values(1,1,1,1,1),(2,2,2,2,2),(3,3,3,3,3),(4,4,4,4,4),(5,5,5,5,5);"
+    qt_sql "select * from ${tableName1} order by k1;"
+    sql "insert into ${tableName2} values(1),(2),(3);"
+    sql "delete from ${tableName1} A using ${tableName2} B where A.k1=B.k;"
+    qt_sql "select * from ${tableName1} order by k1;"
+    sql "set skip_delete_sign=true;"
+    sql "set skip_storage_engine_merge=true;"
+    sql "set skip_delete_bitmap=true;"
+    qt_with_delete_sign "select k1,c1,c2,c3,c4,__DORIS_DELETE_SIGN__ from ${tableName1} order by k1,c1,c2,c3,c4,__DORIS_DELETE_SIGN__;"
+}
```
Comment (zhannngchen) [inline regression-test/suites/unique_with_mow_p0/partial_update/test_partial_update_delete.groovy:60]: Add a partial update stream load case to delete data.

**Comment by bobhan1** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.58 seconds
 stream load tsv:          508 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          29.6 seconds inserted 10000000 Rows, about 337K ops/s
 storage size: 17161006017 Bytes

**Comment by bobhan1** [comment]
run buildall

**Review by zhannngchen — APPROVED** [review]
LGTM

**Comment by bobhan1** [comment]
run click bench

**Comment by bobhan1** [comment]
run clickbench

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 44.82 seconds
 stream load tsv:          506 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.2 seconds inserted 10000000 Rows, about 342K ops/s
 storage size: 17169997476 Bytes

**Review by liaoxin01 — APPROVED** [review]
LGTM

**Pull request merged**